### ISSUE TITLE
[java] UseTryWithResources - false negative for explicit close

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#2882]
**Affects PMD Version:** 6.30.0-SNAPSHOT

**Rule:**[UseTryWithResources](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#usetrywithresources)

**Description:**

From https://github.com/pmd/pmd/pull/2871#issuecomment-717848659

**Code Sample demonstrating the issue:**

https://chunk.io/pmd/fc7db65b2c6a46eca4c9a0c3012482e2/diff/checkstyle/index.html#A360

```java
import java.io.*;

public class UseTryWithResources {
    void method() throws Exception {
        final String fileName = "Test";
        final BufferedReader br = new BufferedReader(new InputStreamReader(
                new FileInputStream(fileName), StandardCharsets.UTF_8));
        try { // warning expected: Consider using a try-with-resources statement instead of explicitly closing the resource 
        } finally {
            br.close();
        }
    }
}
```

**Expected outcome:**

*   Is PMD missing to report a violation, where there should be one? -> false-negative

**Running PMD through:** regression-tester

**Comment by mandeep96** [comment]
Hi, I would like to take this issue if that's ok. First time contributing to open source

**Comment by adangel** [comment]
This has been fixed with PMD 7.0.0-rc1.
